### PR TITLE
adds null pointer check for capture group when omitting empty

### PR DESCRIPTION
**Pull request by eginez**

### CODE REVIEW

**Review by emadelwany — APPROVED** [review]

File: engine/src/main/java/nl/inl/blacklab/search/results/CapturedGroupsImpl.java [hunk 1]
```diff
@@ -114,7 +114,7 @@ public Map<String, Span> getMap(Hit hit, boolean omitEmpty) {
             return null;
         Map<String, Span> result = new TreeMap<>(); // TreeMap to maintain group ordering
         for (int i = 0; i < names.size(); i++) {
-            if (!omitEmpty || groups[i].length() > 0)
+            if (groups[i] != null && (!omitEmpty || groups[i].length() > 0))
```
Comment (brendan-lexion) [inline engine/src/main/java/nl/inl/blacklab/search/results/CapturedGroupsImpl.java:117]: Just checking, are null groups distinct from empty groups semantically? If they're the same I think this should be:

`!omitEmpty || (groups[i] != null && groups[i].length() > 0)`

Of course we'll have `omitEmpty` always set so shouldn't matter much for us, just upstream.

Comment (eginez) [reply engine/src/main/java/nl/inl/blacklab/search/results/CapturedGroupsImpl.java:117]: I think those conditions are equivalent, we only get null elements when the flag is set.

Comment (brendan-lexion) [reply engine/src/main/java/nl/inl/blacklab/search/results/CapturedGroupsImpl.java:117]: Good to know, thanks!

**Comment by eginez** [comment]
Nop I was wrong, we are expecting null pointers when omitting is disabled: https://github.com/INL/BlackLab/pull/341, cherry picked the change here: https://github.com/lexionai/BlackLab/commit/c19c6d18a50f6812e8b10ae662d0c427b839a7c7

**Comment by brendan-lexion** [comment]
Thanks for the followup!

**Pull request merged**